src/tests/GfxReplayTest.cpp

```diff
@@ -121,9 +121,13 @@ TEST(GfxReplayTest, player) {
   // retrieve last child of scene root node
   auto& rootNode = sceneGraph.getRootNode();
   const auto* lastRootChild = rootNode.children().first();
-  ASSERT(lastRootChild);
-  while (lastRootChild->nextSibling()) {
-    lastRootChild = lastRootChild->nextSibling();
+  if (lastRootChild == NULL) {
+    lastRootChild = rootNode;
```

Comment:
@eundersander :
Maybe it is a naive question. I did not quite understand the `lastRootChild`. What does it do? 
(I am not so sure if Vivian is doing the right thing here.)

Reply:
Locally, I'm trying to do the following:
```
if (!lastRootChild) {
    lastRootChild = &rootNode;
  } else {
    ASSERT(lastRootChild);
    while (lastRootChild->nextSibling()) {
      lastRootChild = lastRootChild->nextSibling();
    }
  }
```
This works, but later I fail the test here:
```
else if (keyframeIndex == 0) {
      // assert that a new node was created under root
      ASSERT(lastRootChild->nextSibling());
```

Do you have any feedback on understanding this test?

Reply:
>I did not quite understand the lastRootChild. What does it do?
The scene graph has a root node, and the root node in general has a list of N children. lastRootChild is meant to be a pointer to the last child of the root node. The ordering is due to the order in which children are added to the scene graph.

This test doesn't make any assumption about what exactly that last child is, but it does assume that there is at least N=1 children of the root node upon reaching this point in the code. It seems that Vivian's change is causing that assumption to no longer hold (N=0 for some case), hence her change here. That is fine.

Later, I use lastRootChild to test whether an *additional* child of the root node (i.e., does number of children now equal N + 1?). In the case of N=0, the test needs to be done differently.

One way to clean all this up would be to write a helper function, getNumberOfChildrenOfRoot(). You can modify my tests to just verify that the count is increased by 1 where expected.

Reply:
That makes sense, thanks for the clarification!